### PR TITLE
Add exception mechanism to 'switched' vm

### PR DESCRIPTION
**Pull request by ncos**
- Add 'ret' statement (but still no 'call' statement)
- Add function call stack (exceptions are treated as calls)
- Add a couple of tests

### CODE REVIEW

File: switched.c [hunk 1]
```diff
@@ -127,16 +172,26 @@ int main(int argc, char **argv) {
...
-    
-    cpu_t cpu = {.pc = 0, .sp = -1, .state = Cpu_Running, 
+
+    cpu_t cpu = {.pc = 0, .sp = -1, .csp = -1, .state = Cpu_Running,
                  .steps = 0, .stack = {0},
                  .pmem = Program};
-    
-    while (cpu.state == Cpu_Running && cpu.steps < steplimit) {
+
+    // First, handle pre-enty-point code (like exception handlers)
```
Comment (grigory-rechistov) [inline switched.c]: Would you please comment on why this constraint was introduced? Why must the first instruction always be a jump?

Comment (ncos) [inline switched.c]: This jump is kind of 'interrupt table', but for exceptions. The user will specify the address where to look for exception handlers. This address can not be a constant, as we store exception handler code in program memory, and we do not know neither the size of the program, nor the size of exception handler.

**Comment by ncos** [comment]
- Add new instruction to set exception handler address
- Remove the necessity to start with jump

**Comment by grigory-rechistov** [comment]
Accepted, this closes #3. Kept in the mitrokhin-task03 branch.

**Pull request closed without merging**